framework/src/variables/MooseVariableData.C

```diff
@@ -1794,6 +1803,24 @@ MooseVariableData<OutputType>::fetchADDoFValues()
     if (do_derivatives)
       Moose::derivInsert(_ad_dof_values[i].derivatives(), _dof_indices[i], 1.);
     assignADNodalValue(_ad_dof_values[i], i);
+
+    if (_need_ad_u_dot)
+    {
+      if (_time_integrator && _time_integrator->dt())
+      {
+        _ad_dofs_dot[i] = _ad_dof_values[i];
+        _time_integrator->computeADTimeDerivatives(_ad_dofs_dot[i],
+                                                   _dof_indices[i],
+                                                   _need_ad_u_dotdot ? _ad_dofs_dotdot[i]
+                                                                     : _ad_real_dummy);
+      }
+      // Executing something with a time derivative at initial should not put a NaN
+      else if (_time_integrator && !_time_integrator->dt())
+        _ad_dofs_dot[i] = 0;
+      else
+        mooseError("AD nodal time derivatives not implemented for variables without a time "
+                   "integrator (auxiliary variables)");
+    }
```

Comment:
`computeNodalValues` is the only caller of `fetchADDoFValues`. I've always thought we should sync the names better. As it is, I think it's a bit confusing

Reply:
ok I ll rename!